Use `LayoutUnit::fromRawValue` more in `LayoutUnit.h` and other fixes

Ahmad-S792 opened the pull request:
#### 778c00cc651d4811b2137f2bade7849ca04017e1
<pre>
Use `LayoutUnit::fromRawValue` more in `LayoutUnit.h` and other fixes

<a href="https://bugs.webkit.org/show_bug.cgi?id=285362">https://bugs.webkit.org/show_bug.cgi?id=285362</a>
<a href="https://rdar.apple.com/142330759">rdar://142330759</a>

Reviewed by Antti Koivisto.

Inspired by: <a href="https://source.chromium.org/chromium/chromium/src/+/6477aa568ab9445ef2c5cb1314e237c8c8faeb4c">https://source.chromium.org/chromium/chromium/src/+/6477aa568ab9445ef2c5cb1314e237c8c8faeb4c</a>,
<a href="https://source.chromium.org/chromium/chromium/src/+/54c3bf5d8173d5e11a32cfcb088cf85c818e992a">https://source.chromium.org/chromium/chromium/src/+/54c3bf5d8173d5e11a32cfcb088cf85c818e992a</a> and
<a href="https://source.chromium.org/chromium/chromium/src/+/e22d0d6c01c78bfa7362f9bc57eb4a46256c69ff">https://source.chromium.org/chromium/chromium/src/+/e22d0d6c01c78bfa7362f9bc57eb4a46256c69ff</a>

This patch leverages `fromRawValue` throughout code to reduce duplicate pattern
and it also brings consistency of `static constexpr` throughout code.

Only two function which are made `constexpr` are `epsilon` and `fromRawValue`.

Additionally, it makes four global variables `inline constexpr` as well to take single
copy by using C++17 inline variables.

* Source/WebCore/platform/LayoutUnit.h:
(WebCore::LayoutUnit::fromFloatCeil):
(WebCore::LayoutUnit::fromFloatFloor):
(WebCore::LayoutUnit::fromRawValue):
(WebCore::LayoutUnit::abs const):
(WebCore::LayoutUnit::fraction const):
(WebCore::LayoutUnit::epsilon):
(WebCore::LayoutUnit::max):
(WebCore::LayoutUnit::min):
(WebCore::LayoutUnit::nearlyMax):
(WebCore::LayoutUnit::nearlyMin):
(WebCore::boundedMultiply):
(WebCore::operator/):
(WebCore::operator+):
(WebCore::operator-):
(WebCore::intMod):
(WebCore::operator%):
(WTF::HashTraits&lt;WebCore::LayoutUnit&gt;::emptyValue):

Canonical link: <a href="https://commits.webkit.org/289048@main">https://commits.webkit.org/289048@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/73d7747e6305cde5ac895d77681384dc36b6d7f7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84158 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3779 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38462 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89234 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35167 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86243 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3867 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11750 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65458 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23292 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87204 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2896 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76476 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45751 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2841 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30705 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34215 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73773 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31467 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90614 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11425 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8289 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73909 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11651 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72301 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73113 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18282 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17422 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2770 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11377 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16853 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11226 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14702 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12999 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/platform/LayoutUnit.h:
```diff
@@ -59,10 +59,10 @@ while (0)
-static const int kLayoutUnitFractionalBits = 6;
+static constexpr int kLayoutUnitFractionalBits = 6;
 static constexpr int kFixedPointDenominator = 1 << kLayoutUnitFractionalBits;
-const int intMaxForLayoutUnit = INT_MAX / kFixedPointDenominator;
```
I don't think the `static` are needed here.

Ahmad-S792 replied:
@weinig - Done.

Ahmad-S792 replied:
@weinig - ping?

Ahmad-S792 committed:
Use `LayoutUnit::fromRawValue` more in `LayoutUnit.h` and other fixes

https://bugs.webkit.org/show_bug.cgi?id=285362
rdar://142330759

Reviewed by Antti Koivisto.

Inspired by: https://source.chromium.org/chromium/chromium/src/+/6477aa568ab9445ef2c5cb1314e237c8c8faeb4c,
https://source.chromium.org/chromium/chromium/src/+/54c3bf5d8173d5e11a32cfcb088cf85c818e992a and
https://source.chromium.org/chromium/chromium/src/+/e22d0d6c01c78bfa7362f9bc57eb4a46256c69ff

This patch leverages `fromRawValue` throughout code to reduce duplicate pattern
and it also brings consistency of `static constexpr` throughout code.

Only two function which are made `constexpr` are `epsilon` and `fromRawValue`.

Additionally, it makes four global variables `inline constexpr` as well to take single
copy by using C++17 inline variables.

* Source/WebCore/platform/LayoutUnit.h:
(WebCore::LayoutUnit::fromFloatCeil):
(WebCore::LayoutUnit::fromFloatFloor):
(WebCore::LayoutUnit::fromRawValue):
(WebCore::LayoutUnit::abs const):
(WebCore::LayoutUnit::fraction const):
(WebCore::LayoutUnit::epsilon):
(WebCore::LayoutUnit::max):
(WebCore::LayoutUnit::min):
(WebCore::LayoutUnit::nearlyMax):
(WebCore::LayoutUnit::nearlyMin):
(WebCore::boundedMultiply):
(WebCore::operator/):
(WebCore::operator+):
(WebCore::operator-):
(WebCore::intMod):
(WebCore::operator%):
(WTF::HashTraits<WebCore::LayoutUnit>::emptyValue):

Canonical link: https://commits.webkit.org/289048@main
